### DISCUSSION TITLE
Configurable glob patterns

### DISCUSSION THREAD

**Discussion by DerHannes85** [justkey007/tsc-alias#149] [Ideas]
## Problem

We need to add pure typescript files to a package, so we can use it in another project, where then the build will happen. The source works with `compilerOptions.paths`. Beside of some other preparations, we want to resolve these Path aliases in the packed `*.ts` and `.vue` files. Unfortunately the `globPattern` in [src/index.ts#L48](https://github.com/justkey007/tsc-alias/blob/master/src/index.ts#L48) does not include these file types.

> This is also related to #27 

## Solutions

Make the the `globPattern` in [src/index.ts#L48](https://github.com/justkey007/tsc-alias/blob/master/src/index.ts#L48) configurable so that one can decide which file types are treated.

## Question 

Are you guys interested in such a feature? 

We could fork and implement this, but we would love to see this then later back in your tool.

**Comment by raouldeheer** [comment]
@DerHannes85 Would commit d26254c solve this issue? 
It adds two config options that allow the user to change the glob pattern used for scanning files and change the list of extensions used to check if a path resolves.

Simple example tsconfig: 
```json
{
  ...
  "tsc-alias": {
    "fileExtensions": {
      "inputGlob": "{mjs,cjs,js,jsx,d.{mts,cts,ts,tsx},vue}",
      "outputCheck": [
        "js",
        "json",
        "jsx",
        "cjs",
        "mjs",
        "d.ts",
        "d.tsx",
        "d.cts",
        "d.mts",
        "vue"
      ],
    }
  }
}
```

**Reply by DerHannes85** [reply]
@raouldeheer Wow, thanks for the quick answer and solution. This is more or less exactly what we had in mind. 

~I have added one comment to the PR (`fileExtension: string[]` with `join`). I think this will make it a bit easier to configure and read.~

Another comment would be, if we should also add this to the CLI command as a new option. ~There we then can work with a split to get the `string[]`.~

**Reply by raouldeheer** [reply]
Added the cli options in commit 874d808

**Reply by DerHannes85** [reply]
@raouldeheer Great work. What are the next steps?

**Reply by raouldeheer** [reply]
@DerHannes85 New feature available in `1.8.0`

**Reply by DerHannes85** [reply]
Thank you very much.
